accession-commons-monotonic-generator-jpa/src/test/java/uk/ac/ebi/ampt2d/commons/accession/persistence/jpa/monotonic/service/ContiguousIdBlockServiceTest.java

```diff
@@ -229,14 +231,16 @@ public void testNextBlockWithStartingPointOtherThanZero() {
     }
 
     @Test
-    public void testBlocksWithDuplicateCategoryAndFirstValue() throws DataIntegrityViolationException {
+    public void testBlocksWithDuplicateCategoryAndFirstValue() {
         ContiguousIdBlock block1 = new ContiguousIdBlock(CATEGORY_ID, INSTANCE_ID, 100, 1000);
         repository.save(block1);
-
-        thrown.expect(DataIntegrityViolationException.class);
+        entityManager.flush();
```

Comment:
if this is needed here, have you checked that it's not needed in the main code? I expect the main code is wrapping the saves in an exception, but I haven't checked

Reply:
That is the reason for this other ticket - https://www.ebi.ac.uk/panda/jira/browse/EVA-1851  - to recover from block allocation failures when the non-uniqueness of CATEGORY_ID, FIRST_VALUE is violated.